meson.build

```diff
@@ -87,8 +87,15 @@ configure_file(output : 'libsmp-config.h',
 libsmp_config_incdir = include_directories('.')
 
 if get_option('use-arduino-lib')
-    arduino_core_dep = subproject('arduino-core').get_variable('arduino_core_dep')
-    dependencies = [arduino_core_dep]
+    if run_command('[', '-d', meson.build_root() + '/subprojects/arduino-core', ']').returncode() == 0
+        arduino_core_dep = subproject('arduino-core').get_variable('arduino_core_dep')
+        dependencies = [arduino_core_dep]
+    elif run_command('[', '-d', meson.build_root() +  '/subprojects/teensy-core', ']').returncode() == 0
+        teensy_core_dep = subproject('teensy-core').get_variable('teensy_core_dep')
+        dependencies = [teensy_core_dep]
```

Comment:
Use dependency() function with appropriate flags instead of running a bash command to check subproject